### PR TITLE
add tests to add period and trainer controller

### PR DESCRIPTION
**Pull request by Arturszmer**

### CODE REVIEW

File: src/main/java/com/example/serki/DTO/OfferDTO.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package com.example.serki.DTO;
+
+import java.time.LocalDate;
+import java.util.List;
+
+public class OfferDTO {
```
Comment (Arturszmer) [inline src/main/java/com/example/serki/DTO/OfferDTO.java:6]: make immutable

File: src/main/java/com/example/serki/DTO/PeriodDTO.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
+package com.example.serki.DTO;
+
+import java.time.LocalDate;
+
+public class PeriodDTO {
+
+    private final LocalDate startTraining;
```
Comment (Arturszmer) [inline src/main/java/com/example/serki/DTO/PeriodDTO.java:7]: instant?

File: src/main/java/com/example/serki/DTO/TrainerDTO.java [hunk 3]
```diff
@@ -1,18 +1,40 @@
...
         this.bio=bio;
     }
 
+    public TrainerDTO(String name, String bio, List<LocalDate> unavailableDays) {
+        this.name = name;
+        this.bio = bio;
+        this.unavailableDays = unavailableDays;
+    }
+
+    public TrainerDTO() {
```
Comment (Arturszmer) [inline src/main/java/com/example/serki/DTO/TrainerDTO.java:24]: why this empty constructor?

File: src/main/java/com/example/serki/DTO/TrainerDTO.java [hunk 4]
```diff
@@ -1,18 +1,40 @@
...
     public String setName(){return name;}
 
     public String getBio(){ return bio;}
     public String setBio(){return bio;}
+
+    public List<LocalDate> getUnavailableDays() {
+        return unavailableDays;
+    }
+
+    public void setUnavailableDays(List<LocalDate> unavailableDays) {
```
Comment (Arturszmer) [inline src/main/java/com/example/serki/DTO/TrainerDTO.java:37]: let's try to do without setters

File: src/main/java/com/example/serki/DTO/TypeOfTrainingDTO.java [hunk 5]
```diff
@@ -9,14 +9,28 @@ public class TypeOfTrainingDTO {
...
         this.price = price;
         this.duration = duration;
         this.description = description;
+        this.frontId = frontId;
         this.trainers = trainers;
+        this.periodDTOS = periodDTOS;
+
+    }
+
+    public TypeOfTrainingDTO() {
```
Comment (Arturszmer) [inline src/main/java/com/example/serki/DTO/TypeOfTrainingDTO.java:33]: not needed?

File: src/main/java/com/example/serki/Exceptions/TrainingPeriodExist.java [hunk 6]
```diff
@@ -0,0 +1,7 @@
+package com.example.serki.Exceptions;
+
+public class TrainingPeriodExist extends RuntimeException {
```
Comment (Arturszmer) [inline src/main/java/com/example/serki/Exceptions/TrainingPeriodExist.java:3]: TrainingPeriodExistException ?

File: src/main/java/com/example/serki/models/TypeOfTraining.java [hunk 7]
```diff
@@ -13,26 +13,39 @@ public class TypeOfTraining {
...
+    private List<TrainingPeriod> trainingPeriod;
     @ManyToMany
     private List<Trainer> trainer;
 
-    public TypeOfTraining(String name, double price, double duration, String description, List<Trainer> trainer) {
+    public TypeOfTraining(String name,
+                          double price,
+                          double duration,
+                          String description,
+                          String frontId,
```
Comment (Arturszmer) [inline src/main/java/com/example/serki/models/TypeOfTraining.java:27]: id?

File: src/main/java/com/example/serki/models/TypeOfTraining.java [hunk 8]
```diff
@@ -97,7 +106,10 @@ public int hashCode() {
         return Objects.hash(id, name, price, duration, description, trainer);
     }
 
-    public void assign(Trainer trainer) {
+    public void assignTrainer(Trainer trainer) {
```
Comment (Arturszmer) [inline src/main/java/com/example/serki/models/TypeOfTraining.java:109]: assign was also ok

File: src/main/java/com/example/serki/service/TypeOfTrainingService.java [hunk 9]
```diff
@@ -68,13 +63,13 @@ public void addTrainerToTraining(TrainerAssignmentDTO trainerAssignmentDTO){
         TypeOfTraining typeOfTraining = typeOfTrainingsRepo.findByName(trainerAssignmentDTO.getTypeOfTrainingName())
                 .orElseThrow(TypeOfTrainingNotExist::new);
 
-        isTrainerAsigned(trainerAssignmentDTO, typeOfTraining);
+        isTrainerAssigned(trainerAssignmentDTO, typeOfTraining);
```
Comment (Arturszmer) [inline src/main/java/com/example/serki/service/TypeOfTrainingService.java:66]: result of the method is not used

File: src/main/java/com/example/serki/DTO/OfferDTO.java (hunk 1 above)
Comment (Arturszmer) [reply src/main/java/com/example/serki/DTO/OfferDTO.java:6]: done

File: src/main/java/com/example/serki/service/TypeOfTrainingService.java (hunk 9 above)
Comment (Arturszmer) [reply src/main/java/com/example/serki/service/TypeOfTrainingService.java:66]: it checks is trainer assignment already exist

**Review by SlorenisBP — APPROVED** [review]

**Pull request merged**